src/main/kotlin/oop/State/SemaphoreState.kt

```diff
@@ -0,0 +1,12 @@
+package oop.State
+
+sealed class SemaphoreState {
+
+  class GreenSemaphore : SemaphoreState()
```

Comment:
You are not right, is an `object` when you want only one instance of this class. But I'm wondering  in my last resolution is it's good a `object` instance. (`object` improve the performance because we only manage types and not does `new` instances every time in my next state. 

have a look this when Raul Raja speak about `Option` and `object None`